react_app/src/components/Game/ChallengeWidget.js

```diff
@@ -22,6 +22,8 @@ class ChallengeWidget extends React.Component {
     this.state = {
       open: true
     };
+    // think i already mentioned this, but this implementation of a dialog
+    // is really confusing - there's much easier ways!
```

Comment:
I accept that what I'm doing here is a little weird, but I wanted the functionality of a material-ui dialog, although it appears to be built to be bound 1 to 1 to a single button. Perhaps I need to rethink my choice of component instead of attempting to fit square peg into round hole?